app/controllers/posts_controller.rb

```diff
@@ -1,9 +1,8 @@
 class PostsController < ApplicationController
-  before_action :authenticate_user!
+  before_action :authenticate_user!, except: %i[index show]
   def index
+    @posts = Post.where(author_id: params[:user_id])
     @user = User.find(params[:user_id])
```

Comment:
- [ ] Looks like currently you are not authorising posts while it's in the project descriptions. There is a statement that you can use to do this,kindly look at the aid below :point_down: 
```
  class PostsController < ApplicationController
    load_and_authorize_resource

    ...
  end
```
That statement  applies your cancancan rules :wink: